mParticle-ROKOMobi/MPKitROKOMobi.m

```diff
@@ -0,0 +1,207 @@
+//
+//  MPKitROKOMobi.m
+//
+//  Copyright 2017 ROKO Labs, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import "MPKitROKOMobi.h"
+
+@interface MPKitROKOMobiProxy: NSObject <MPKitROKOMobiProvider>
+
+@property (nonatomic, strong) ROKOInstaBot *instabot;
+@property (nonatomic, strong) ROKOLinkManager *linkManager;
+
+@end
+
+@implementation MPKitROKOMobiProxy
+
+- (ROKOInstaBot *)getInstaBot {
+    @synchronized (self) {
+        if (!_instabot) {
+            _instabot = [ROKOInstaBot new];
+        }
+        return _instabot;
+    }
+}
+
+- (ROKOLinkManager *)getLinkManager {
+    @synchronized (self) {
+        if (!_linkManager) {
+            _linkManager = [ROKOLinkManager new];
+        }
+        return _linkManager;
+    }
+}
+
+@end
+
+@interface MPKitROKOMobi() <ROKOLinkManagerDelegate>
+
+@property (nonatomic, strong) ROKOPush *pusher;
+@property (nonatomic, strong) ROKOLinkManager *linkManager;
+@property (nonatomic, strong) id <MPKitROKOMobiProvider> proxy;
+
+@end
+
+@implementation MPKitROKOMobi
+
+- (id <MPKitROKOMobiProvider>)proxy {
+    @synchronized (self) {
+        if (!_proxy) {
+            _proxy = [[MPKitROKOMobiProxy alloc] init];
+        }
+        return _proxy;
+    }
+}
++ (NSNumber *)kitCode {
+    return @123;
+}
+
++ (void)load {
+    MPKitRegister *kitRegister = [[MPKitRegister alloc] initWithName:@"ROKOMobi" className:@"MPKitROKOMobi" startImmediately:YES];
+    [MParticle registerExtension:kitRegister];
+}
+
+#pragma mark - MPKitInstanceProtocol methods
+
+#pragma mark Kit instance and lifecycle
+- (nonnull instancetype)initWithConfiguration:(nonnull NSDictionary *)configuration startImmediately:(BOOL)startImmediately {
+    self = [super init];
+    NSString *appKey = configuration[@"apiKey"];
```

Comment:
appkey is never used anywhere - should it be used to initialize the SDK?

Reply:
yes, we will update PR to pass in the Instabot API key for SDK initialization from the mParticle wrapper so it can be set server-side.